apollo-router/src/configuration/mod.rs

```diff
@@ -416,16 +417,31 @@ impl Configuration {
     pub(crate) fn rust_query_planner_config(
         &self,
     ) -> apollo_federation::query_plan::query_planner::QueryPlannerConfig {
-        apollo_federation::query_plan::query_planner::QueryPlannerConfig {
+        use apollo_federation::query_plan::query_planner::QueryPlanIncrementalDeliveryConfig;
+        use apollo_federation::query_plan::query_planner::QueryPlannerConfig;
+        use apollo_federation::query_plan::query_planner::QueryPlannerDebugConfig;
+
+        let max_evaluated_plans = self
+            .supergraph
+            .query_planning
+            .experimental_plans_limit
+            // Fails if experimental_plans_limit is zero; use our default.
+            .and_then(NonZeroU32::new)
+            .unwrap_or(NonZeroU32::new(10_000).expect("it is not zero"));
+
+        QueryPlannerConfig {
             reuse_query_fragments: self.supergraph.reuse_query_fragments.unwrap_or(true),
             subgraph_graphql_validation: false,
             generate_query_fragments: self.supergraph.generate_query_fragments,
-            incremental_delivery:
-                apollo_federation::query_plan::query_planner::QueryPlanIncrementalDeliveryConfig {
-                    enable_defer: self.supergraph.defer_support,
-                },
+            incremental_delivery: QueryPlanIncrementalDeliveryConfig {
+                enable_defer: self.supergraph.defer_support,
+            },
             type_conditioned_fetching: self.experimental_type_conditioned_fetching,
-            debug: Default::default(),
+            debug: QueryPlannerDebugConfig {
+                bypass_planner_for_single_subgraph: false,
```

Comment:
we probably also pass this down ([link](https://github.com/apollographql/router/blob/dev/apollo-federation/src/query_plan/query_planner.rs#L370))?

Reply:
there is no router config to enable this option as far as i know. my general feeling is that the option could be removed entirely from the query planner, but 🤷🏻‍♀️

Reply:
Hmm it is something that we do support in the gateway (which I believe was done as an experimental feature to skip the expensive QP computation) - I'll open a ticket to figure out what we want to do on RS side of things

Reply:
I think it's risky to support this if in the future we move support for authentication directives etc. into the query planner